### PR TITLE
ISPN-12140 OS service CA encryption

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-12140

### CODE REVIEW

File: documentation/asciidoc/topics/ref_encryption_service_ca.adoc [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+If the {openshift} service CA is available, {ispn_operator} adds the following
+`spec.security.endpointEncryption` configuration to the **Infinispan** CR:
+
+[source,options="nowrap",subs=attributes+]
+----
+spec:
+  ...
+  security:
+    endpointEncryption:
+      type: service
```
Comment (dmvolod) [inline documentation/asciidoc/topics/ref_encryption_service_ca.adoc:16]: @rigazilla maybe we can change the endpointEncryption type to the CamelCase in order with other k8s and ocp? entities?

File: documentation/asciidoc/topics/ref_encryption_service_ca.adoc [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      certServiceName: service.beta.openshift.io <1>
+      certSecretName: {example_crd_name}-cert-secret <2>
+----
+
+<1> Specifies the {openshift} Service.
+<2> Names the secret that contains a service certificate, `tls.crt`, and key, `tls.key`, in PEM format. By default, {ispn_operator} appends `-cert-secret` to the {brandname} cluster name.
```
Comment (dmvolod) [inline documentation/asciidoc/topics/ref_encryption_service_ca.adoc:22]: @oraNod maybe a little bit clear related default: If this name is not defined, {ispn_operator} expects following secret name _{brandname} cluster name_ `-cert-secret`.

File: documentation/asciidoc/topics/ref_encryption_service_ca.adoc (hunk 1 above)
Comment (dmvolod) [reply documentation/asciidoc/topics/ref_encryption_service_ca.adoc:16]: We can use compare ignore case in code, but implement aligning with other k8s kind names.

Comment (rigazilla) [reply documentation/asciidoc/topics/ref_encryption_service_ca.adoc:16]: @dmvolod ignore case compare is here: https://github.com/infinispan/infinispan-operator/pull/479

Comment (dmvolod) [reply documentation/asciidoc/topics/ref_encryption_service_ca.adoc:16]: Cool, @rigazilla , thanks, will review

Comment (dmvolod) [reply documentation/asciidoc/topics/ref_encryption_service_ca.adoc:16]: @oraNod looks like we also need to fix endpointEncryption.type in proc_encrypting_endpoints_tls_secret.adoc to CamelCase

Comment (oraNod) [reply documentation/asciidoc/topics/ref_encryption_service_ca.adoc:16]: @dmvolod thanks. I've changed it throughout so `type: CamelCase` - let me know if I've misunderstood the change.

File: documentation/asciidoc/topics/ref_services.adoc [hunk 3]
```diff
@@ -60,7 +60,7 @@ Allows {brandname} to back up data between clusters in different locations.
 | Protocol
 | Description
 
-| `$NAME-sites`
+| `<name>-sites`
```
Comment (rigazilla) [inline documentation/asciidoc/topics/ref_services.adoc:63]: Reading through the code this seems to be
```suggestion
| `<cluster_name>-site`
```

File: documentation/asciidoc/topics/ref_encryption_service_ca.adoc (hunk 1 above)
Comment (dmvolod) [reply documentation/asciidoc/topics/ref_encryption_service_ca.adoc]: Sorry, @oraNod maybe I'm not so correctly wrote.
Camel Case style writing for type
https://en.m.wikipedia.org/wiki/Camel_case

Comment (oraNod) [reply documentation/asciidoc/topics/ref_encryption_service_ca.adoc]: @dmvolod ok. isn't it correct that the first letter for any value of the `type` field needs to be caps? so `type: Value` not `type: value`

sorry if I'm making a meal of it at this stage. just getting a little confused bouncing between stuff.

Comment (dmvolod) [reply documentation/asciidoc/topics/ref_encryption_service_ca.adoc]: @oraNod , I'm not sure about all types, but for any `kind` type need to be started with uppercase

Comment (oraNod) [reply documentation/asciidoc/topics/ref_encryption_service_ca.adoc]: thanks @dmvolod I think we should be good here in that case

**Comment by rigazilla** [comment]
merged thanks @oraNod

**Pull request merged**